gadget/update.go

```diff
@@ -119,3 +119,16 @@ func canUpdateStructure(from *PositionedStructure, to *PositionedStructure) erro
 
 	return nil
 }
+
+func canUpdateVolume(from *PositionedVolume, to *PositionedVolume) error {
+	if from.ID != to.ID {
+		return fmt.Errorf("cannot change volume ID from %q to %q", from.ID, to.ID)
+	}
+	if from.EffectiveSchema() != to.EffectiveSchema() {
+		return fmt.Errorf("cannot change volume schema from %q to %q", from.EffectiveSchema(), to.EffectiveSchema())
+	}
+	if len(from.PositionedStructure) != len(to.PositionedStructure) {
```

Comment:
Curious, why cannot we change the number of structures?

Reply:
To fully support changing the number of structure would mean we'd need to allow adding or removing partition on the disk during update. We may allow this in future, but not in the initial version.